### PR TITLE
Us 30/fronted library implemtation

### PR DESCRIPTION
**Pull request by victor-villca**
# What did I do?
Implemented the java code for the frontend library based on the UML diagram.
image

# How did I do it?
Following the UML diagram.

# Why did I do it?
To create a functioning library for our games to follow.

### CODE REVIEW

File: inputsuscription/EventType.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
```
Comment (zrock-dev) [inline inputsuscription/EventType.java:16]: Please remove this method, the EventType classes will be read only.

File: inputsuscription/Handler.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        subscribers.get(event).add(subscriber);  
+    }
+
+
+    public void notifySubscribers(EventType eventType) {
+        if (subscribers.containsKey(eventType)) {
+            for (ISubscriber subscriber : subscribers.get(eventType)) {
+                subscriber.notify(eventType);
+            }
+        }
```
Comment (zrock-dev) [inline inputsuscription/Handler.java:30]: Please log as an error when the condition is not met.

**Review by EmanuelGalindoC — APPROVED** [review]
It looks nice

File: piece/Sound.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    private String path;
+
+    public boolean canPlay() {
+        return false;
+    }
+
+    public void setPath(String path) {
+        this.path = path;
+    }
```
Comment (zrock-dev) [inline piece/Sound.java:13]: Remove this method. And create a constructor with the path as an argument.

File: piece/Sound.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package piece;
+
+public abstract class Sound implements IFile {
+
+    private String path;
+
+    public boolean canPlay() {
+        return false;
+    }
```
Comment (zrock-dev) [inline piece/Sound.java:9]: this is an abstract method

**Review by zrock-dev — CHANGES_REQUESTED** [review]
Nice job, please append the image of the UML. I've left some commentaries.

**Pull request merged**